modules/ols-providing-custom-knowledge-to-the-llm.adoc

```diff
@@ -0,0 +1,111 @@
+[id="providing-custom-knowledge-to-the-llm_{context}"]
+= Providing custom knowledge to the LLM
+
+Customize the information available to the large language model (LLM) by providing access to a container image that resides in a remote image registry. The examples in this procedure use `quay.io` as the remote container image registry, and the path for the custom image is `quay.io/<username>/my-byok-image:latest`.
+
+:FeatureName: The BYO Knowledge feature
+include::snippets/technology-preview.adoc[]
+
+.Prerequisites
+
+* You are logged in to the {ocp-product-title} web console as a user with the `cluster-admin` role. Alternatively, you are logged in to a user account that has permission to create a cluster-scoped CR file.
+
+* You have installed the {ols-long} Operator.
+
+* You have configured the large language model (LLM) provider.
+
+* The custom information you want to add resides as a collection of Markdown files; no other file format is supported.
+
+* You have an account for a container image registry, such as `quay.io`.
+
+.Procedure 
+
+. Specify the location of the directory with the markdown files for the custom RAG database and the path for the image that the BYO Knowledge tool generates by running the following command: 
++
+[source,terminal]
+----
+$ podman run -e OUT_IMAGE_TAG=my-byok-image -it --rm --device=/dev/fuse -v <directory_for_markdown_files>:/markdown:Z -v <directory_for_image_tar>:/output:Z $MY_BYOK_TOOL_IMAGE
+----
++
+[NOTE]
+====
+The container image that the tool generates contains a custom RAG database. The RAG database provides additional information to the LLM. 
+====
+
+. Load the container image that the BYO Knowledge tool generated by running the following command:
++
+[source,terminal]
+----
+$ podman load < <directory_for_image_tar>/<my-byok-image.tar>
+----
+
+. Display the podman images that reside on your laptop by running the following command:
++
+[source,terminal]
+----
+$ podman images
+----
++
+.Example output
+[source,terminal]
+----
+REPOSITORY                            TAG                IMAGE ID      CREATED       SIZE
+localhost/my-byok-image               latest             be7d1770bf10  1 minute  ago    2.37 GB
+...
+----
+
+. Tag the local image with a new name and destination so that the image can be pushed to the container image registry by running the following command:
++
+[source,terminal]
+----
+$ podman tag localhost/my-byok-image:latest quay.io/<username>/my-byok-image:latest
+----
+
+. Push the local container image to the container image registry by running the following command:
++
+[source,terminal]
+----
+$ podman push quay.io/<username>/my-byok-image:latest
+----
+
+. Modify the `OLSconfig` custom resource to deploy the newly created RAG database alongside the existing one:
+
+.. In the {ocp-product-title} web console, click *Operators* -> *Installed Operators*. 
+
+.. Select *All Projects* in the  *Project* dropdown at the top of the screen.
+
+.. Click {ols-long} Operator.
```

Comment:
Should the title be bold as a UI item in this case?
```suggestion
.. Click *{ols-long} Operator*.
```

Reply:
done